agent/xds/routes.go

```diff
@@ -167,6 +204,21 @@ func (s *ResourceGenerator) makeRoutes(
 	return resources, nil
 }
 
+func (s *ResourceGenerator) makeRoutesForAddresses(name string, addresses map[string]string, autoHostRewrite bool) ([]proto.Message, error) {
+
+	var resources []proto.Message
+	var lb *structs.LoadBalancer
+
+	route, err := makeNamedAddressesRouteWithLB(name, addresses, lb, autoHostRewrite)
+	if err != nil {
+		s.Logger.Error("failed to make route", "cluster", "error", err)
+		return nil, err
+	}
+	resources = append(resources, route)
+
+	return resources, nil
+}
+
```

Comment:
Any reason for the indirection with this function to call `makeNamedAddressesRouteWithLB`? Do we need any of the LoadBalancer logic?